### ISSUE TITLE
plugins/parsers/json: tag_keys can not be applied to sub-objects

### ISSUE THREAD

**Issue by igomura** [influxdata/telegraf#6853]
### Relevant telegraf.conf:
```toml
[[inputs.file]]
files = ["data.json"]
data_format = "json"
tag_keys = [
  "Serial",
  "Model",
  "Firmware",
]
```

### System info:
Linux CentOS 7.7
Telegraf v1.12.6

### Steps to reproduce:

Input Data:
```json
{
  "device0": {
    "Count": 0,
    "Errors": 0,
    "Serial": "9JHNGTUT",
    "Model": "WDC WUH721414ALE604",
    "Firmware": "LDGSW07G"
  },
  "device1": {
    "Count": 0,
    "Errors": 0,
    "Serial": "9JHLPW9T",
    "Model": "WDC WUH721414ALE604",
    "Firmware": "LDGSW07G"
  }
}
```

### Expected behavior:

file,Firmware=LDGSW07G,Model=WDC\ WUH721414ALE604,Serial=9JHNGTUT device0_Count=0,device0_Errors=0 1578088300000000000
file,Firmware=LDGSW07G,Model=WDC\ WUH721414ALE604,Serial=9JHLPW9T device1_Errors=0,device1_Count=0 1578088300000000000

### Actual behavior:

file device0_Count=0,device1_Count=0,device0_Errors=0,device1_Errors=0 1578088240000000000

### Additional info:

In opposite, data below will pickup tags but there is no place for path which can be important key

```json
[
  {
    "Count": 0,
    "Errors": 0,
    "Serial": "9JHNGTUT",
    "Model": "WDC WUH721414ALE604",
    "Firmware": "LDGSW07G"
  },
  {
    "Count": 0,
    "Errors": 0,
    "Serial": "9JHLPW9T",
    "Model": "WDC WUH721414ALE604",
    "Firmware": "LDGSW07G"
  }
]
```

file,Firmware=LDGSW07G,Model=WDC\ WUH721414ALE604,Serial=9JHLPW9T Errors=0,Count=0 1578088310000000000
file,Firmware=LDGSW07G,Model=WDC\ WUH721414ALE604,Serial=9JHNGTUT Count=0,Errors=0 1578088310000000000

**Comment by danielnelson** [comment]
Probably the most Telegraf friendly way this can be done today is to put the device name in the object using your second form, and then add it as a tag_key.

```toml
[[inputs.file]]
  files = ["data.json"]
  data_format = "json"
  tag_keys = [
    "Serial",
    "Model",
    "Firmware",
    "Device"
  ]
```

```
[
  {
    "Device": "device0",
    "Count": 0,
    "Errors": 0,
    "Serial": "9JHNGTUT",
    "Model": "WDC WUH721414ALE604",
    "Firmware": "LDGSW07G"
  },
  {
    "Device": "device1",
    "Count": 0,
    "Errors": 0,
    "Serial": "9JHLPW9T",
    "Model": "WDC WUH721414ALE604",
    "Firmware": "LDGSW07G"
  }
]
```

**Comment by igomura** [comment]
This is what I'm doing now. But in fact, my JSON is more complicated. It looks like the one below.
In this situation I have to decide should I keep tags or devices hierarchy. 

```json
{
  "Group A": [
    {
      "Sub-group 1": [
        {
          "Count": 0,
          "Errors": 0,
          "Serial": "9JHNGTUT",
          "Model": "WDC WUH721414ALE604",
          "Firmware": "LDGSW07G"
        }
      ]
    }
  ],
  "Group B": [
    {
      "Sub-group 1": [
        {
          "Count": 0,
          "Errors": 0,
          "Serial": "9JHLPW9T",
          "Model": "WDC WUH721414ALE604",
          "Firmware": "LDGSW07G"
        }
      ]
    }
  ]
}
```

**Comment by danielnelson** [comment]
We could make a change to allow globs in the tag_keys, similar to what was done recently for string fields.  I think that in your case though this will give you less than ideal results since the field names will contain the index making it hard to query.

So, if you have control over the input data, maybe you can instead continue to push down the values:

```json
[
	{
		"Group": "Group A",
		"SubGroup": "Sub-group 1",
		"Count": 0,
		"Errors": 0,
		"Serial": "9JHNGTUT",
		"Model": "WDC WUH721414ALE604",
		"Firmware": "LDGSW07G"
	},
	{
		"Group": "Group B",
		"SubGroup": "Sub-group 1",
		"Count": 0,
		"Errors": 0,
		"Serial": "9JHLPW9T",
		"Model": "WDC WUH721414ALE604",
		"Firmware": "LDGSW07G"
	}
]
```

Or flatten the path into a single key:
```json
[
	{
		"Path": "/Group A/Sub-group 1",
		"Count": 0,
		"Errors": 0,
		"Serial": "9JHNGTUT",
		"Model": "WDC WUH721414ALE604",
		"Firmware": "LDGSW07G"
	},
	{
		"Path": "/Group B/Sub-group 1",
		"Count": 0,
		"Errors": 0,
		"Serial": "9JHLPW9T",
		"Model": "WDC WUH721414ALE604",
		"Firmware": "LDGSW07G"
	}
]
```

**Comment by igomura** [comment]
It can be solved this way in my case. But each object may have its own metrics and tags. The whole json hierarchy becomes useless as it has to be flatten in order to apply tags. IMHO, It would be better to pickup metrics and tags from each level of the hierarchy instead.

**Comment by sjwang90** [comment]
The new [JSON_v2 parser](https://github.com/influxdata/telegraf/tree/master/plugins/parsers/json_v2) (`json_v2`) has just been released in [Telegraf 1.19 rc-1](https://github.com/influxdata/telegraf/releases/tag/v1.19.0-rc1). Please download and test it out with your JSON files and open any bugs or feature gaps you may encounter.
